auth/api/iam/openid4vp.go

```diff
@@ -81,8 +87,33 @@ func (r Wrapper) handleAuthorizeRequestFromHolder(ctx context.Context, verifier
 	}
 	// now we have a valid redirectURL, so all future errors will redirect to this URL using the Oauth2ErrorWriter
 
+	// quick check for missing string values that otherwise would blow up
+	for _, param := range []string{oauth.ClientIDParam, oauth.ScopeParam, oauth.StateParam} {
+		if _, ok := params[param].(string); !ok {
+			return nil, withCallbackURI(oauthError(oauth.InvalidRequest, fmt.Sprintf("missing %s parameter", param)), redirectURL)
+		}
+	}
+
+	// additional JAR checks
+	// check if the audience is the verifier
+	audience, ok := params[jwt.AudienceKey].([]string)
+	if !ok || len(audience) == 0 || audience[0] != verifier.String() {
```

Comment:
if we only check the first entry, maybe assure that the length is exactly 1, instead of not empty? Then at least it breaks whne behavior becomes unspecified.

Reply:
fixed with  `singleStringParam`